src/signals/models.py

```diff
@@ -0,0 +1,237 @@
+
+
+class TimeTypeChoices(models.TextChoices):
+    """
+    A class representing choices for time types.
+    """
+    DAY = 'day', _('Day')
+    WEEK = 'week', _('Week')
+
+
+class TimeLabelChoices(models.TextChoices):
+    """
+    A class representing choices for time labels.
+    """
+    DAY = 'day', _('Day')
+    DATE = 'week', _('Week')
+
+
+class FormatChoices(models.TextChoices):
+    """
+    A class representing choices for formats.
+    """
+    RAW = 'raw', _('Raw')
+    PERCENT = 'percent', _('Percent')
+    FRACTION = 'fraction', _('Fraction')
+    COUNT = 'count', _('Count')
+    PER100K = 'per100k', _('Per 100K')
+
+
+class HighValuesAreChoices(models.TextChoices):
+    """
+    A class representing choices for high values.
+    """
+    BAD = 'bad', _('Bad')
+    GOOD = 'good', _('Good')
+    NEUTRAL = 'neutral', _('Neutral')
+
+
+class SignalCategory(models.Model):
+    """
+    A model representing a signal category.
+    """
+    name = models.CharField(
+        help_text=_('Name'),
+        max_length=128,
+        unique=True
+    )
+
+    def __str__(self) -> str:
+        """
+        Returns the name of the signal category as a string.
+
+        :return: The name of the signal category as a string.
+        :rtype: str
+        """
+        return self.name
+
+
+class SignalType(models.Model):
+    """
+    A model representing a signal type.
+    """
+    name = models.CharField(
+        help_text=_('Name'),
+        max_length=128,
+        unique=True
+    )
+
+    def __str__(self) -> str:
+        """
+        Returns the name of the signal type as a string.
+
+        :return: The name of the signal type as a string.
+        :rtype: str
+        """
+        return self.name
+
+
+class Pathogen(models.Model):
+    """
+    A model representing a pathogen.
+    """
+    name = models.CharField(
+        help_text=_('Name'),
+        max_length=128,
+        unique=True
+    )
+
+    def __str__(self) -> str:
+        """
+        Returns the name of the pathogen as a string.
+
+        :return: The name of the pathogen as a string.
+        :rtype: str
+        """
+        return self.name
+
+
+class Geography(models.Model):
+    """
+    A model representing a available geography.
+    """
+    name = models.CharField(
+        help_text=_('Name'),
+        max_length=128,
+        unique=True
+    )
+
+    class Meta:
+        verbose_name_plural = "geographies"
+
+    def __str__(self) -> str:
+        """
+        Returns the name of the available geography as a string.
+
+        :return: The name of the available geography as a string.
+        :rtype: str
+        """
+        return self.name
+
+
+class Signal(models.Model):
+    """
+    A model representing a signal.
+    """
+    name = models.CharField(
+        help_text=_('Name'),
+        max_length=128,
+        unique=True
+    )
+    base = models.ForeignKey(
+        'signals.Signal',
+        help_text=_('Signal base'),
```

Comment:
is this useful or no?

```suggestion
        related_name='base_for',
        help_text=_('Signal base'),
```